RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -0,0 +1,127 @@
+
+    public static class BulkResponse extends RestClient.Response {
+        public BulkResponse(int responseCode, String responseBody, String responseMessage) {
+            super(responseCode, responseBody, responseMessage);
+        }
+
+        public boolean hasBadStatusCode() {
+            return !(code == HttpURLConnection.HTTP_OK || code == HttpURLConnection.HTTP_CREATED);
+        }
+
+        public boolean hasFailedOperations() {
+            return body.contains("\"errors\":true");
+        }
+
+        public String getFailureMessage() {
+            String failureMessage;
+            if (hasBadStatusCode()) {
+                failureMessage = "Bulk request failed.  Status code: " + code + ", Response body: " + body;
+            } else {
+                failureMessage = "Bulk request succeeded, but some operations failed.  Response body: " + body;
+            }
+
+            return failureMessage;
+        }
+    }
+
+    public final ConnectionDetails connectionDetails;
+    private final RestClient client;
+
+    public OpenSearchClient(ConnectionDetails connectionDetails) {
+        this.connectionDetails = connectionDetails;
+        this.client = new RestClient(connectionDetails);
+    }
+
+    /*
+     * Create a legacy template if it does not already exist; return true if created, false otherwise.
+     */
+    public boolean createLegacyTemplateIdempotent(String templateName, ObjectNode settings){
+        String targetPath = "_template/" + templateName;
+        return createObjectIdempotent(targetPath, settings);
+    }
+
+    /*
+     * Create a component template if it does not already exist; return true if created, false otherwise.
+     */
+    public boolean createComponentTemplateIdempotent(String templateName, ObjectNode settings){
+        String targetPath = "_component_template/" + templateName;
+        return createObjectIdempotent(targetPath, settings);
+    }
+
+    /*
+     * Create an index template if it does not already exist; return true if created, false otherwise.
+     */
+    public boolean createIndexTemplateIdempotent(String templateName, ObjectNode settings){
+        String targetPath = "_index_template/" + templateName;
+        return createObjectIdempotent(targetPath, settings);
+    }
+
+    /*
+     * Create an index if it does not already exist; return true if created, false otherwise.
+     */
+    public boolean createIndexIdempotent(String indexName, ObjectNode settings){
+        String targetPath = indexName;
+        return createObjectIdempotent(targetPath, settings);
+    }
+
+    private boolean createObjectIdempotent(String objectPath, ObjectNode settings){
+        RestClient.Response response = client.get(objectPath, true);
+        if (response.code == HttpURLConnection.HTTP_NOT_FOUND) {
+            client.put(objectPath, settings.toString(), false);
+            return true;
+        } else if (response.code == HttpURLConnection.HTTP_OK) {
```

Comment:
Is this the right status code?  It doesn't seem like it from the log line below

Reply:
It should be the correct status code, as it's for the GET not the PUT.